app/models/concerns/metadata.rb

```diff
@@ -0,0 +1,11 @@
+module Metadata
+  extend ActiveSupport::Concern
+
+  included do
+    attribute :metadata, ActiveRecord::Type::Json.new
+
+    def metadata
```

Comment:
@hsong-rh 
One issue we have here is we need to be able to add extra data into the metadata other than just the user_capabilities. For example in catalog we add extra properties like 
'shared' => true|false

If we have to do this then we have to override the metadata method in the included class name.

https://github.com/RedHatInsights/catalog-api/blob/40e8e8749f5aef59abb0864e4e1acfb8d8549544/app/models/portfolio.rb#L24

Should the mixin be called user_capabilities instead of metadata?